### DISCUSSION TITLE
Why are the "fuel-asm", "fuel-types", and "fuel-tx" crates duplicated in this repository?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [FuelLabs/fuel-vm#295] [Q&A] [answered]
In the [CONTRIBUTING.md](https://github.com/FuelLabs/fuel-core/blob/a3f584f12bb3e160aa00e08616a27c81734ebd02/CONTRIBUTING.md) guide of the `fuel-core` repository, we are told that:

> The Client team actively develops and maintains several dependencies used in Fuel Core, which you may be also interested in:
> - [fuel-types](https://github.com/FuelLabs/fuel-types)
> - [fuel-merkle](https://github.com/FuelLabs/fuel-merkle)
> - [fuel-tx](https://github.com/FuelLabs/fuel-tx)
> - [fuel-asm](https://github.com/FuelLabs/fuel-asm)
> - [fuel-vm](https://github.com/FuelLabs/fuel-vm)

So I find it a bit confusing that I find `fuel-asm`, `fuel-types`, and `fuel-tx` existing as crates part of this repository (`fuel-vm`), as well as independent crates  living in separate repositories.

What is the most up-to-date version of these crates, and why are they duplicated?

**Comment by adlerjohn** [comment, marked as answer]
https://github.com/FuelLabs/fuel-vm/issues/269

Issue ☝️ needs more description, but TL;DR those are the repos from before the monorepo migration. They'll be archived shortly. `fuel-vm` will contain the monorepo of those crates.
